### PR TITLE
Support emptyDir for volumes and infer dnsPolicy from hostNetwork

### PR DESCRIPTION
**Pull request by jiacheliu3**
This resolves:
1. Feature request https://github.com/Alluxio/alluxio/issues/11555
1. Comments under feature request https://github.com/Alluxio/alluxio/issues/11526#issuecomment-641687729

### CODE REVIEW

**Comment by jiacheliu3** [comment]
@cheyang I wasn't able to add you as a reviewer. Please take a look. Thanks!

File: integration/kubernetes/helm-chart/alluxio/templates/_helpers.tpl [hunk 1]
```diff
@@ -257,6 +257,24 @@ resources:
...
+  {{- if eq .Values.shortCircuit.volumeType "hostPath" }}
+        - name: alluxio-domain
+          hostPath:
+            path: {{ .Values.shortCircuit.hostPath }}
+            type: DirectoryOrCreate
+  {{- else if eq .Values.shortCircuit.volumeType "persistentVolumeClaim" }}
+        - name: alluxio-domain
+          persistentVolumeClaim:
+            claimName: "{{ .Values.shortCircuit.pvcName }}"
+  {{- else }}
```
Comment (cheyang) [inline integration/kubernetes/helm-chart/alluxio/templates/_helpers.tpl:270]: I think empty dir for alluxio-domain is not helpful. The reason is that empty dir is only shared in the same pod, not across different pods. So it's not visible to Fuse pod or other client pod.

Comment (jiacheliu3) [reply integration/kubernetes/helm-chart/alluxio/templates/_helpers.tpl]: Makes sense to me. Removed emptyDir for domain socket and keeping only PVC and hostPath

**Comment by jiacheliu3** [comment]
@madanadit This change has been validated by @cheyang 's group in their environment.

File: integration/kubernetes/helm-chart/alluxio/templates/fuse/daemonset.yaml [hunk 2]
```diff
@@ -92,7 +94,7 @@ spec:
               fieldRef:
                 fieldPath: status.hostIP
           - name: ALLUXIO_CLIENT_JAVA_OPTS
-            value: " -Dalluxio.user.hostname=$(ALLUXIO_CLIENT_HOSTNAME) "
+            value: " -Dalluxio.user.hostname=${ALLUXIO_CLIENT_HOSTNAME} "
```
Comment (jiacheliu3) [inline integration/kubernetes/helm-chart/alluxio/templates/fuse/daemonset.yaml:97]: Fixed one typo

File: integration/kubernetes/helm-chart/alluxio/values.yaml [hunk 3]
```diff
@@ -70,18 +70,14 @@ master:
     embedded: 19200
     rpc: 19998
     web: 19999
-  # You can specify hostNetwork and dnsPolicy here to override the default
-  # hostNetwork: false
+  hostNetwork: false
+  # dnsPolicy will be ClusterFirstWithHostNet if hostNetwork: true
+  # and ClusterFirst if hostNetwork: false
+  # You can specify dnsPolicy here to override this inference
```
Comment (jiacheliu3) [inline integration/kubernetes/helm-chart/alluxio/values.yaml:76]: Per https://github.com/Alluxio/alluxio/issues/11526#issuecomment-641687729, inferring dnsPolicy from hostNetwork, but not inferring hostNetwork from whether it's root.

File: integration/kubernetes/helm-chart/alluxio/values.yaml [hunk 4]
```diff
@@ -230,14 +247,20 @@ shortCircuit:
...
   policy: uuid
-  # Attributes if the domain socket volume is PVC
+  # volumeType controls the type of shortCircuit volume.
+  # It can be "persistentVolumeClaim" or "hostPath"
+  volumeType: persistentVolumeClaim
+  size: 1Mi
+  # Attributes to use if the domain socket volume is PVC
   pvcName: alluxio-worker-domain-socket
-  pvcAccessModes:
+  accessModes:
```
Comment (madanadit) [inline integration/kubernetes/helm-chart/alluxio/values.yaml:256]: Please note this and any other renames in changelog. A rename will break back compatibility, avoid if not absolutely necessary

Comment (jiacheliu3) [reply integration/kubernetes/helm-chart/alluxio/values.yaml:256]: Actually i found this `accessModes` i'm changing to is the one used in the docs. I realized this is changed to `pvcAccessModes` sometime in 2.3 snapshot. So I think it's okay.

Comment (madanadit) [reply integration/kubernetes/helm-chart/alluxio/values.yaml:256]: yep

**Comment by madanadit** [comment]
alluxio-bot, merge this please.

**Pull request merged**